capstone-frontend/src/components/ClubPage.jsx

```diff
@@ -1,10 +1,27 @@
 import React, { Component } from 'react';
 
 class ClubPage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      response: {}
+    }
+  }
+
+  fetchClubData = () => {
+    fetch(`/api/clubs?id=${this.props.match.params.id}`)
+        .then(response => response.json()).then(res => this.setState({response: res[0]}));
+  }
+
   render() {
+    this.fetchClubData();
```

Comment:
This should probably be in `componentDidMount` (and also probably `componentDidUpdate` in case `match.params.id` changes). Those are generally the two places you want to make API calls. Keep logic and side-effects out of your `render` method whenever possible. It also looks to me like there might be an infinite `render` -> `fetchClubData` -> `setState` -> `render`... loop here: you can verify that by looking at the Network tab in the Dev Tools. If so, using `componentDidMount` should fix that since it only gets invoked once.

Reply:
Good idea, I'll create both those functions and have them call ``` fetchClubData() ```.